src/source/deprecatedSource.ts

```diff
@@ -0,0 +1,4 @@
+/**
+ * @deprecated This class should not be used to create a Source.
+ */
+export class Source {}
```

Comment:
The `@deprecated` mention is more meant to be used as "this is still working, but we're gonna axe it in the next major"

I would remove the file altogether